### ISSUE TITLE
How to train on coco dataset?

### ISSUE THREAD

**Issue by Yc174** [CharlesShang/TFFRCNN#72]
I see that code have coco file,so how to train on coco dataset? @CharlesShang @eragonruan Thanks a lot.

**Comment by Yc174** [comment]
I have modified some files so that I almost could train on coco dataset, but some errors had happened. 
The info as follows:
/home/yanchao/anaconda2/bin/python /home/yanchao/TFFRCNN/faster_rcnn/train_net.py
kittivoc_train
kittivoc_val
kittivoc_trainval
kittivoc_test
kittivoc_train
kittivoc_val
kittivoc_trainval
kittivoc_test
kittivoc_train
kittivoc_val
kittivoc_trainval
kittivoc_test
nthu_71
nthu_370
usage: train_net.py [-h] [--gpu GPU_ID] [--solver SOLVER] [--iters MAX_ITERS]
                    [--weights PRETRAINED_MODEL] [--cfg CFG_FILE]
                    [--imdb IMDB_NAME] [--rand] [--network NETWORK_NAME]
                    [--set ...] [--restore RESTORE]

Train a Fast R-CNN network

optional arguments:
  -h, --help            show this help message and exit
  --gpu GPU_ID          GPU device id to use [0]
  --solver SOLVER       solver prototxt
  --iters MAX_ITERS     number of iterations to train
  --weights PRETRAINED_MODEL
                        initialize with pretrained model weights
  --cfg CFG_FILE        optional config file
  --imdb IMDB_NAME      dataset to train on
  --rand                randomize (do not use a fixed seed)
  --network NETWORK_NAME
                        name of the network
  --set ...             set config keys
  --restore RESTORE     restore or not
Using config:
{'ANCHOR_SCALES': [8, 16, 32],
 'DATA_DIR': '/home/yanchao/TFFRCNN/data',
 'DEDUP_BOXES': 0.0625,
 'EPS': 1e-14,
 'EXP_DIR': 'faster_rcnn_coco_vgg',
 'GPU_ID': 0,
 'IS_EXTRAPOLATING': True,
 'IS_MULTISCALE': False,
 'IS_RPN': True,
 'LOG_DIR': 'faster_rcnn_coco_vgg',
 'MATLAB': 'matlab',
 'MODELS_DIR': '/home/yanchao/TFFRCNN/models/pascal_voc',
 'NCLASSES': 81,
 'NET_NAME': 'VGGnet',
 'PIXEL_MEANS': array([[[ 102.9801,  115.9465,  122.7717]]]),
 'REGION_PROPOSAL': 'RPN',
 'RNG_SEED': 3,
 'ROOT_DIR': '/home/yanchao/TFFRCNN',
 'SUBCLS_NAME': 'voxel_exemplars',
 'TEST': {'BBOX_REG': True,
          'HAS_RPN': True,
          'MAX_SIZE': 1000,
          'NMS': 0.3,
          'PROPOSAL_METHOD': 'selective_search',
          'RPN_MIN_SIZE': 16,
          'RPN_NMS_THRESH': 0.7,
          'RPN_POST_NMS_TOP_N': 300,
          'RPN_PRE_NMS_TOP_N': 6000,
          'SCALES': [600],
          'SVM': False},
 'TRAIN': {'ASPECTS': [1],
           'ASPECT_GROUPING': True,
           'BATCH_SIZE': 300,
           'BBOX_INSIDE_WEIGHTS': [1, 1, 1, 1],
           'BBOX_NORMALIZE_MEANS': [0.0, 0.0, 0.0, 0.0],
           'BBOX_NORMALIZE_STDS': [0.001, 0.001, 0.002, 0.002],
           'BBOX_NORMALIZE_TARGETS': True,
           'BBOX_NORMALIZE_TARGETS_PRECOMPUTED': True,
           'BBOX_REG': True,
           'BBOX_THRESH': 0.5,
           'BG_THRESH_HI': 0.5,
           'BG_THRESH_LO': 0.0,
           'DISPLAY': 10,
           'DONTCARE_AREA_INTERSECTION_HI': 0.5,
           'FG_FRACTION': 0.3,
           'FG_THRESH': 0.5,
           'GAMMA': 0.1,
           'HAS_RPN': True,
           'IMS_PER_BATCH': 1,
           'KERNEL_SIZE': 5,
           'LEARNING_RATE': 0.001,
           'LOG_IMAGE_ITERS': 100,
           'MAX_SIZE': 1000,
           'MOMENTUM': 0.9,
           'OHEM': True,
           'PRECLUDE_HARD_SAMPLES': True,
           'PROPOSAL_METHOD': 'gt',
           'RANDOM_DOWNSAMPLE': False,
           'RPN_BATCHSIZE': 256,
           'RPN_BBOX_INSIDE_WEIGHTS': [1, 1, 1, 1],
           'RPN_CLOBBER_POSITIVES': False,
           'RPN_FG_FRACTION': 0.5,
           'RPN_MIN_SIZE': 16,
           'RPN_NEGATIVE_OVERLAP': 0.3,
           'RPN_NMS_THRESH': 0.7,
           'RPN_POSITIVE_OVERLAP': 0.7,
           'RPN_POSITIVE_WEIGHT': -1.0,
           'RPN_POST_NMS_TOP_N': 2000,
           'RPN_PRE_NMS_TOP_N': 12000,
           'SCALES': [600],
           'SCALES_BASE': [0.25, 0.5, 1.0, 2.0, 3.0],
           'SNAPSHOT_INFIX': '',
           'SNAPSHOT_ITERS': 5000,
           'SNAPSHOT_PREFIX': 'VGGnet_fast_rcnn',
           'SOLVER': 'Momentum',
           'STEPSIZE': 60000,
           'USE_FLIPPED': True,
           'USE_PREFETCH': False,
           'WEIGHT_DECAY': 0.0005},
 'USE_GPU_NMS': True}
<bound method coco.default_roidb of <lib.datasets.coco.coco object at 0x7f47b1180890>>
loading annotations into memory...
Done (t=9.62s)
creating index...
index created!
Loaded dataset `coco_2014_train` for training
Appending horizontally-flipped training examples...
coco_2014_train gt roidb loaded from /home/yanchao/TFFRCNN/data/cache/coco_2014_train_gt_roidb.pkl
done
Preparing training data...
done
Output will be saved to `/home/yanchao/TFFRCNN/output/faster_rcnn_coco_vgg/coco_2014_train`
Logs will be saved to `/home/yanchao/TFFRCNN/logs/faster_rcnn_coco_vgg/coco_2014_train/2017-06-06-22-37-30`
/gpu:0
Tensor("data:0", shape=(?, ?, ?, 3), dtype=float32)
Tensor("conv5_3/Relu:0", shape=(?, ?, ?, 512), dtype=float32)
Tensor("rpn_conv/3x3/Relu:0", shape=(?, ?, ?, 512), dtype=float32)
Tensor("rpn_conv/3x3/Relu:0", shape=(?, ?, ?, 512), dtype=float32)
Tensor("rpn_cls_score/BiasAdd:0", shape=(?, ?, ?, 18), dtype=float32)
Tensor("gt_boxes:0", shape=(?, 5), dtype=float32)
Tensor("gt_ishard:0", shape=(?,), dtype=int32)
Tensor("dontcare_areas:0", shape=(?, 4), dtype=float32)
Tensor("im_info:0", shape=(?, 3), dtype=float32)
Tensor("rpn_cls_score/BiasAdd:0", shape=(?, ?, ?, 18), dtype=float32)
Tensor("rpn_cls_prob:0", shape=(?, ?, ?, ?), dtype=float32)
Tensor("Reshape_2:0", shape=(?, ?, ?, 18), dtype=float32)
Tensor("rpn_bbox_pred/BiasAdd:0", shape=(?, ?, ?, 36), dtype=float32)
Tensor("im_info:0", shape=(?, 3), dtype=float32)
Tensor("rpn_rois:0", shape=(?, 5), dtype=float32)
Tensor("gt_boxes:0", shape=(?, 5), dtype=float32)
Tensor("gt_ishard:0", shape=(?,), dtype=int32)
Tensor("dontcare_areas:0", shape=(?, 4), dtype=float32)
Tensor("conv5_3/Relu:0", shape=(?, ?, ?, 512), dtype=float32)
Tensor("roi-data/rois:0", shape=(?, 5), dtype=float32)
[<tf.Tensor 'conv5_3/Relu:0' shape=(?, ?, ?, 512) dtype=float32>, <tf.Tensor 'roi-data/rois:0' shape=(?, 5) dtype=float32>]
Tensor("drop7/mul:0", shape=(?, 4096), dtype=float32)
Use network `VGGnet_train` in training
2017-06-06 22:37:30.575890: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.
2017-06-06 22:37:30.575904: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
2017-06-06 22:37:30.575907: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
2017-06-06 22:37:30.575909: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX2 instructions, but these are available on your machine and could speed up CPU computations.
2017-06-06 22:37:30.575911: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use FMA instructions, but these are available on your machine and could speed up CPU computations.
2017-06-06 22:37:30.675171: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:901] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2017-06-06 22:37:30.675520: I tensorflow/core/common_runtime/gpu/gpu_device.cc:887] Found device 0 with properties: 
name: GeForce GTX TITAN X
major: 5 minor: 2 memoryClockRate (GHz) 1.076
pciBusID 0000:01:00.0
Total memory: 11.92GiB
Free memory: 11.63GiB
2017-06-06 22:37:30.675530: I tensorflow/core/common_runtime/gpu/gpu_device.cc:908] DMA: 0 
2017-06-06 22:37:30.675533: I tensorflow/core/common_runtime/gpu/gpu_device.cc:918] 0:   Y 
2017-06-06 22:37:30.675537: I tensorflow/core/common_runtime/gpu/gpu_device.cc:977] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GTX TITAN X, pci bus id: 0000:01:00.0)
Computing bounding-box regression targets...
bbox target means:
[[ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]
 [ 0.  0.  0.  0.]]
[ 0.  0.  0.  0.]
bbox target stdevs:
[[ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]
 [ 0.001  0.001  0.002  0.002]]
[ 0.001  0.001  0.002  0.002]
Normalizing targets
done
Solving...
/home/yanchao/anaconda2/lib/python2.7/site-packages/tensorflow/python/ops/gradients_impl.py:93: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.
  "Converting sparse IndexedSlices to a dense Tensor of unknown shape. "
Loading pretrained model weights from /home/yanchao/TFFRCNN/data/pretrain_model/VGG_imagenet.npy
assign pretrain model weights to conv5_1
assign pretrain model biases to conv5_1
assign pretrain model weights to fc6
assign pretrain model biases to fc6
assign pretrain model weights to conv5_3
assign pretrain model biases to conv5_3
assign pretrain model weights to fc7
assign pretrain model biases to fc7
ignore fc8
ignore fc8
assign pretrain model weights to conv5_2
assign pretrain model biases to conv5_2
assign pretrain model weights to conv4_1
assign pretrain model biases to conv4_1
assign pretrain model weights to conv4_2
assign pretrain model biases to conv4_2
assign pretrain model weights to conv4_3
assign pretrain model biases to conv4_3
assign pretrain model weights to conv3_3
assign pretrain model biases to conv3_3
assign pretrain model weights to conv3_2
assign pretrain model biases to conv3_2
assign pretrain model weights to conv3_1
assign pretrain model biases to conv3_1
assign pretrain model weights to conv1_1
assign pretrain model biases to conv1_1
assign pretrain model weights to conv1_2
assign pretrain model biases to conv1_2
assign pretrain model weights to conv2_2
assign pretrain model biases to conv2_2
assign pretrain model weights to conv2_1
assign pretrain model biases to conv2_1
Tensor("im_info:0", shape=(?, 3), dtype=float32)
Tensor("gt_boxes:0", shape=(?, 5), dtype=float32)
iter: 0 / 200000, total loss: 22.2288, rpn_loss_cls: 1.3745, rpn_loss_box: 0.3967, loss_cls: 3.9417, loss_box: 16.5159, lr: 0.001000
speed: 2.363s / iter

image: COCO_train2014_000000248009.jpg Tensor("im_info:0", shape=(?, 3), dtype=float32)
Tensor("gt_boxes:0", shape=(?, 5), dtype=float32)
Tensor("im_info:0", shape=(?, 3), dtype=float32)
Tensor("gt_boxes:0", shape=(?, 5), dtype=float32)
iter: 10 / 200000, total loss: 81.1020, rpn_loss_cls: 1.3110, rpn_loss_box: 0.8646, loss_cls: 4.4321, loss_box: 74.4943, lr: 0.001000
speed: 2.121s / iter
image: COCO_train2014_000000408078.jpg Tensor("im_info:0", shape=(?, 3), dtype=float32)
Tensor("gt_boxes:0", shape=(?, 5), dtype=float32)
Tensor("im_info:0", shape=(?, 3), dtype=float32)
Tensor("gt_boxes:0", shape=(?, 5), dtype=float32)
iter: 20 / 200000, total loss: 77.8428, rpn_loss_cls: 1.3875, rpn_loss_box: 0.5267, loss_cls: 1.4938, loss_box: 74.4348, lr: 0.001000
speed: 0.293s / iter
image: COCO_train2014_000000557602.jpg Tensor("im_info:0", shape=(?, 3), dtype=float32)
Tensor("gt_boxes:0", shape=(?, 5), dtype=float32)
Tensor("im_info:0", shape=(?, 3), dtype=float32)
Tensor("gt_boxes:0", shape=(?, 5), dtype=float32)
iter: 30 / 200000, total loss: 18.3304, rpn_loss_cls: 1.2020, rpn_loss_box: 4.3564, loss_cls: 0.4864, loss_box: 12.2856, lr: 0.001000
speed: 0.294s / iter
image: COCO_train2014_000000452342.jpg Tensor("im_info:0", shape=(?, 3), dtype=float32)
Tensor("gt_boxes:0", shape=(?, 5), dtype=float32)
Tensor("im_info:0", shape=(?, 3), dtype=float32)
Tensor("gt_boxes:0", shape=(?, 5), dtype=float32)
iter: 40 / 200000, total loss: 27.4952, rpn_loss_cls: 1.3475, rpn_loss_box: 0.5339, loss_cls: 0.5553, loss_box: 25.0585, lr: 0.001000
speed: 0.287s / iter
image: COCO_train2014_000000468626.jpg Tensor("im_info:0", shape=(?, 3), dtype=float32)
Tensor("gt_boxes:0", shape=(?, 5), dtype=float32)
Tensor("im_info:0", shape=(?, 3), dtype=float32)
Tensor("gt_boxes:0", shape=(?, 5), dtype=float32)
iter: 50 / 200000, total loss: 47.5796, rpn_loss_cls: 1.2731, rpn_loss_box: 0.7431, loss_cls: 0.6044, loss_box: 44.9589, lr: 0.001000
speed: 0.825s / iter
image: COCO_train2014_000000467647.jpg Tensor("im_info:0", shape=(?, 3), dtype=float32)
Tensor("gt_boxes:0", shape=(?, 5), dtype=float32)
Tensor("im_info:0", shape=(?, 3), dtype=float32)
Tensor("gt_boxes:0", shape=(?, 5), dtype=float32)
iter: 60 / 200000, total loss: 52.0918, rpn_loss_cls: 1.1619, rpn_loss_box: 2.5039, loss_cls: 1.6238, loss_box: 46.8023, lr: 0.001000
speed: 1.575s / iter
image: COCO_train2014_000000536164.jpg Tensor("im_info:0", shape=(?, 3), dtype=float32)
Tensor("gt_boxes:0", shape=(?, 5), dtype=float32)
Tensor("im_info:0", shape=(?, 3), dtype=float32)
Tensor("gt_boxes:0", shape=(?, 5), dtype=float32)
iter: 70 / 200000, total loss: 19.4080, rpn_loss_cls: 1.1959, rpn_loss_box: 0.3126, loss_cls: 0.4042, loss_box: 17.4953, lr: 0.001000
speed: 0.328s / iter
image: COCO_train2014_000000004318.jpg Tensor("im_info:0", shape=(?, 3), dtype=float32)
Tensor("gt_boxes:0", shape=(?, 5), dtype=float32)
Tensor("im_info:0", shape=(?, 3), dtype=float32)
Tensor("gt_boxes:0", shape=(?, 5), dtype=float32)
iter: 80 / 200000, total loss: 2.5716, rpn_loss_cls: 0.9210, rpn_loss_box: 1.3939, loss_cls: 0.1321, loss_box: 0.1245, lr: 0.001000
speed: 2.514s / iter
Traceback (most recent call last):
  File "/home/yanchao/anaconda2/lib/python2.7/site-packages/tensorflow/python/ops/script_ops.py", line 82, in __call__
    ret = func(*args)
  File "/home/yanchao/TFFRCNN/faster_rcnn/../lib/rpn_msr/anchor_target_layer_tf.py", line 140, in anchor_target_layer
    argmax_overlaps = overlaps.argmax(axis=1) # (A)
ValueError: attempt to get argmax of an empty sequence
2017-06-06 22:40:41.164622: W tensorflow/core/framework/op_kernel.cc:1152] Internal: Failed to run py callback pyfunc_0: see error log.
2017-06-06 22:40:41.178946: W tensorflow/core/framework/op_kernel.cc:1152] Internal: Failed to run py callback pyfunc_0: see error log.
	 [[Node: rpn-data/PyFunc = PyFunc[Tin=[DT_FLOAT, DT_FLOAT, DT_INT32, DT_FLOAT, DT_FLOAT, DT_INT32, DT_INT32], Tout=[DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT], token="pyfunc_0", _device="/job:localhost/replica:0/task:0/cpu:0"](rpn_cls_score/BiasAdd/_71, _recv_gt_boxes_0, _recv_gt_ishard_0, _recv_dontcare_areas_0, _recv_im_info_0, rpn-data/PyFunc/input_5, rpn-data/PyFunc/input_6)]]
Traceback (most recent call last):
  File "/home/yanchao/TFFRCNN/faster_rcnn/train_net.py", line 132, in <module>
    restore=bool(int(args.restore)))
  File "/home/yanchao/TFFRCNN/faster_rcnn/../lib/fast_rcnn/train.py", line 402, in train_net
    sw.train_model(sess, max_iters, restore=restore)
  File "/home/yanchao/TFFRCNN/faster_rcnn/../lib/fast_rcnn/train.py", line 257, in train_model
    cls_prob, bbox_pred, rois =  sess.run(fetches=fetch_list, feed_dict=feed_dict)
  File "/home/yanchao/anaconda2/lib/python2.7/site-packages/tensorflow/python/client/session.py", line 778, in run
    run_metadata_ptr)
  File "/home/yanchao/anaconda2/lib/python2.7/site-packages/tensorflow/python/client/session.py", line 982, in _run
    feed_dict_string, options, run_metadata)
  File "/home/yanchao/anaconda2/lib/python2.7/site-packages/tensorflow/python/client/session.py", line 1032, in _do_run
    target_list, options, run_metadata)
  File "/home/yanchao/anaconda2/lib/python2.7/site-packages/tensorflow/python/client/session.py", line 1052, in _do_call
    raise type(e)(node_def, op, message)
tensorflow.python.framework.errors_impl.InternalError: Failed to run py callback pyfunc_0: see error log.
	 [[Node: rpn-data/PyFunc = PyFunc[Tin=[DT_FLOAT, DT_FLOAT, DT_INT32, DT_FLOAT, DT_FLOAT, DT_INT32, DT_INT32], Tout=[DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT], token="pyfunc_0", _device="/job:localhost/replica:0/task:0/cpu:0"](rpn_cls_score/BiasAdd/_71, _recv_gt_boxes_0, _recv_gt_ishard_0, _recv_dontcare_areas_0, _recv_im_info_0, rpn-data/PyFunc/input_5, rpn-data/PyFunc/input_6)]]
	 [[Node: gradients/Gather_grad/Size/_87 = _HostRecv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/gpu:0", send_device="/job:localhost/replica:0/task:0/cpu:0", send_device_incarnation=1, tensor_name="edge_1064_gradients/Gather_grad/Size", tensor_type=DT_INT32, _device="/job:localhost/replica:0/task:0/gpu:0"]()]]

Caused by op u'rpn-data/PyFunc', defined at:
  File "/home/yanchao/TFFRCNN/faster_rcnn/train_net.py", line 124, in <module>
    network = get_network(args.network_name)
  File "/home/yanchao/TFFRCNN/faster_rcnn/../lib/networks/factory.py", line 29, in get_network
    return VGGnet_train()
  File "/home/yanchao/TFFRCNN/faster_rcnn/../lib/networks/VGGnet_train.py", line 17, in __init__
    self.setup()
  File "/home/yanchao/TFFRCNN/faster_rcnn/../lib/networks/VGGnet_train.py", line 59, in setup
    .anchor_target_layer(_feat_stride, anchor_scales, name = 'rpn-data' ))
  File "/home/yanchao/TFFRCNN/faster_rcnn/../lib/networks/network.py", line 36, in layer_decorated
    layer_output = op(self, layer_input, *args, **kwargs)
  File "/home/yanchao/TFFRCNN/faster_rcnn/../lib/networks/network.py", line 274, in anchor_target_layer
    [tf.float32,tf.float32,tf.float32,tf.float32])
  File "/home/yanchao/anaconda2/lib/python2.7/site-packages/tensorflow/python/ops/script_ops.py", line 189, in py_func
    input=inp, token=token, Tout=Tout, name=name)
  File "/home/yanchao/anaconda2/lib/python2.7/site-packages/tensorflow/python/ops/gen_script_ops.py", line 40, in _py_func
    name=name)
  File "/home/yanchao/anaconda2/lib/python2.7/site-packages/tensorflow/python/framework/op_def_library.py", line 768, in apply_op
    op_def=op_def)
  File "/home/yanchao/anaconda2/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 2336, in create_op
    original_op=self._default_original_op, op_def=op_def)
  File "/home/yanchao/anaconda2/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 1228, in __init__
    self._traceback = _extract_stack()

InternalError (see above for traceback): Failed to run py callback pyfunc_0: see error log.
	 [[Node: rpn-data/PyFunc = PyFunc[Tin=[DT_FLOAT, DT_FLOAT, DT_INT32, DT_FLOAT, DT_FLOAT, DT_INT32, DT_INT32], Tout=[DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT], token="pyfunc_0", _device="/job:localhost/replica:0/task:0/cpu:0"](rpn_cls_score/BiasAdd/_71, _recv_gt_boxes_0, _recv_gt_ishard_0, _recv_dontcare_areas_0, _recv_im_info_0, rpn-data/PyFunc/input_5, rpn-data/PyFunc/input_6)]]
	 [[Node: gradients/Gather_grad/Size/_87 = _HostRecv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/gpu:0", send_device="/job:localhost/replica:0/task:0/cpu:0", send_device_incarnation=1, tensor_name="edge_1064_gradients/Gather_grad/Size", tensor_type=DT_INT32, _device="/job:localhost/replica:0/task:0/gpu:0"]()]]


Process finished with exit code 1

**Comment by eragonruan** [comment]
there may be something wrong with you annotation file, you can set display=1 to see exactly in which image this error occurs, then check the annotation file of this image, see if everything is ok.

**Comment by Yc174** [comment]
Thank you very much! I have set display=1 and found when the iter=85 and the image=0005500.jpg it will stop and error orrors, but I found it's a normal image with width=480 and height=640. I really don't know where not normal. But I have found  Ross Girshick's code used  @instances_valminusminival2014.json and instances_minival2014.json,which I can't download beaucase that link isn't useful. @eragonruan
